lib/hacluster.pm

```diff
@@ -1137,4 +1138,47 @@ sub check_iscsi_failure {
     }
 }
 
+=head3 cluster_check_status_by_regex
+
+Remotely check 'crm status' output against regular expression
+=over 3
+
+=item B<SHOW_CLUSTER_STATUS> - Output from 'crm status' command
+
+=item B<REGULAR_EXPRESSION> - Regular expression to match the text to find
+
+=back
+=cut
+
+sub cluster_check_status_by_regex {
+    my ($show_cluster_status, $regular_expression) = @_;     
+    croak 'No crm status output' if (!$show_cluster_status);
+    croak 'No regular expression provided' unless (ref $show_cluster_status eq 'Regexp');
+  
+    my $done = 0;   
+    
+    for my $line (split("\n", $show_cluster_status)) {
+        if ($line =~ $regular_expression) {
+            my @cluster_status = ("Stopped", "Failed", "Pending", "Blocked");
+
+            foreach my $cluster_state (@cluster_status) {
+                if ($line =~ $cluster_state) {
+                    $done = 1;
+                    print "$cluster_state matches the text '$line'\n";
+                    last;
+                }
+            }            
+        }
+        if ($done == 1) {
+            record_info("crm status",
+            join("\n------------\n", "Output : $show_cluster_status",
+                "regexp : $regular_expression",
+                "done : $done",));
+            sleep 30 unless $done;
```

Comment:
What's this `sleep 30` needed for? What are you waiting for?

Reply:
removed